src/stores/api/formatters/ethereum.go

```diff
@@ -60,11 +62,55 @@ func FormatSignTypedDataRequest(request *types.SignTypedDataRequest) *signer.Typ
 	return typedData
 }
 
-func FormatTransaction(tx *types.SignETHTransactionRequest) *ethtypes.Transaction {
-	if tx.To == nil {
-		return ethtypes.NewContractCreation(uint64(tx.Nonce), tx.Value.ToInt(), uint64(tx.GasLimit), tx.GasPrice.ToInt(), tx.Data)
+func FormatTransaction(tx *types.SignETHTransactionRequest) (*ethtypes.Transaction, error) {
```

Comment:
In my opinion we should fail in the following cases:
- user set `GasPrice` and it does not use type `DynamicFeeTxType`
- user set `AccessList` and it does use Legacy
- ....

Reply:
the behavior of nodes (the standard) is to ignore the fields that do not make sense given the type of tx. If you provide gasPrice but the tx is EIP-1559 (dynamic fee) then it ignores gasPrice. Same for accessList. Personally I prefer that behavior

Reply:
discussed